docs/asciidoc/advanced-contextPropagation.adoc

```diff
@@ -32,25 +48,80 @@ TL.set("HELLO");
 
 Mono.deferContextual(ctx ->
   Mono.delay(Duration.ofSeconds(1))
-      //we're now in another thread, TL is not set
-      .map(v -> "delayed ctx[" + TLKEY + "]=" + ctx.getOrDefault(TLKEY, "not found") + ", TL=" + TL.get())
-)
-.contextCapture()
-.block(); // returns "delayed ctx[TLKEY]=HELLO, TL=null"
+      //we're now in another thread, TL is not explicitly set
+      .map(v -> "delayed ctx[" + TLKEY + "]=" + ctx.getOrDefault(TLKEY, "not found") + ", TL=" + TL.get()))
+.contextCapture() // can be skipped in automatic mode when a blocking operator follows
+.block(); // returns "delayed ctx[TLKEY]=HELLO, TL=null" in default mode
+          // returns "delayed ctx[TLKEY]=HELLO, TL=HELLO" in automatic mode
+----
+====
+
+NOTE: In the **automatic** mode, blocking operators, such as `Flux#blockFirst()`,
+`Flux#blockLast()`, `Flux#toIterable()`, `Mono#block()`, `Mono#blockOptional()`, and
+relevant overloads, all perform `contextCapture()` transparently, so in most cases it is
+not necessary to add it.
+
+=== `contextWrite` Operator
+
+When the `ThreadLocal` values are not and do not need to be present at the time of
+subscription, we can immediately store the value in the `Context`:
+
+====
+[source,java]
+----
+//assuming TL is known to Context-Propagation as key TLKEY.
+static final ThreadLocal<String> TL = new ThreadLocal<>();
+
+//in the main thread, TL is not set
+
+Mono.deferContextual(ctx ->
+  Mono.delay(Duration.ofSeconds(1))
+      //we're now in another thread, TL is not explicitly set
+      .map(v -> "delayed ctx[" + TLKEY + "]=" + ctx.getOrDefault(TLKEY, "not found") + ", TL=" + TL.get()))
+.contextWrite(ctx -> ctx.put(TLKEY, "HELLO"))
+.block(); // returns "delayed ctx[TLKEY]=HELLO, TL=null" in default mode
+          // returns "delayed ctx[TLKEY]=HELLO, TL=HELLO" in automatic mode
 ----
 ====
 
-== Operators that transparently restore a snapshot: `handle` and `tap`
-Both `Flux` and `Mono` variants of `handle` and `tap` will have their behavior slightly modified
-if the Context-Propagation library is available at runtime.
+== Accessing `ThreadLocal` state corresponding to `ContextView`
+
+As can be seen in the above examples, in the default mode, the `ThreadLocal` value is
+`null` inside the lambda provided to the `map` operator. This shows a major difference
+between the **default** (limited) mode, and the **automatic** mode.
+
+=== Automatic mode
```

Comment:
why do we have automatic and default modes again if the previous sections was talking about default mode already?

Reply:
may be not reference at the beginning about default at all and instead of automatic start with default saying that this mode is available starting from 3.5 and does minimal stuff and then mention automatic which is available in 3.5.9+ and stabilised in 3.6